### PR TITLE
mindustry-server: Fix

### PR DESCRIPTION
**Pull request by chkno**
###### Description

Fix `mindustry-server` build after https://github.com/NixOS/nixpkgs/pull/221396

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
    Can't:

      earlyoom: mem avail:   155 of  7829 MiB ( 1.99%), swap free: 1047 of 10540 MiB ( 9.94%)
      earlyoom: low memory! at or below SIGTERM limits: mem 10.00%, swap 10.00%
      earlyoom: sending SIGTERM to process 490532 uid 1000 "nix-env": badness 1093, VmRSS 5789 MiB

- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/mindustry/default.nix [hunk 1]
```diff
@@ -202,7 +202,6 @@ stdenv.mkDerivation rec {
       --set ALSA_PLUGIN_DIR ${alsa-plugins}/lib/alsa-lib/'' + optionalString enableWayland '' \
       --set SDL_VIDEODRIVER wayland \
       --set SDL_VIDEO_WAYLAND_WMCLASS Mindustry
-    '' + ''
```
Comment (rhendric) [inline pkgs/games/mindustry/default.nix:205]: Oh, oops. No, this isn't quite right either; it makes all the subsequent code conditional on `enableWayland` and not on `enableClient` as intended. We should instead insert an open paren after `enableClient` on line 196 and close paren after the close quote on line 218, I think.

**Review by rhendric — APPROVED** [review]
Haven't tested but LGTM.

File: pkgs/games/mindustry/default.nix [hunk 2]
```diff
@@ -215,11 +213,21 @@ stdenv.mkDerivation rec {
...
     install -Dm644 server/build/libs/server-release.jar $out/share/mindustry-server.jar
     mkdir -p $out/bin
     makeWrapper ${jdk}/bin/java $out/bin/mindustry-server \
       --add-flags "-jar $out/share/mindustry-server.jar"
+  '';
+
+  installPhase = with lib; ''
+    runHook preInstall
+  '' + optionalString enableClient ''
+    runHook installClient
```
[the remark this answers is not in the document]
Comment (chkno) [reply pkgs/games/mindustry/default.nix]: Hrm.  It seems fine to me, but it does indeed look unprecedented.  I agree with the sentiment of being thoughtful around precedents.  This is a simple problem with multiple simple fixes.  It ought not require deep thought.  I'll use a different simple fix.

`eval "$preInstall"` does the same thing as `runHook` without invoking the weighty concept of _Hook_, but is also unprecedented, so I'll not suggest that.

Looking around, I found a slightly better convention for nested conditional string concatenation: Use ${} string interpolation to get new scopes ([example](https://github.com/NixOS/nixpkgs/blob/87f9156865ab09e3bde39aadb4131ae364ae704e/pkgs/development/mobile/xcodeenv/build-app.nix#L96)).  I'd still rather avoid nested conditional string concatenation, though.

--

I've updated this to use a simple `let`.  Thanks.

--

(While looking around, I noticed a few misspelled hook names & [made a PR to fix them](https://github.com/NixOS/nixpkgs/pull/233854).)

**Comment by fgaz** [comment]
@GrahamcOfBorg build mindustry mindustry.tests.nixosTest

**Review by fgaz — APPROVED** [review]

**Pull request merged**